fiboa_cli/datasets/jecam.py

```diff
@@ -0,0 +1,66 @@
+    To facilitate the location of sampling areas and the remote acquisition of waypoints, field operators were equipped with GPS tablets providing access to a QGIS project with Very High Spatial Resolution (VHSR) images ordered just before the surveys. For each waypoint, a set of attributes, corresponding to the cropping practices (crop type, cropping pattern, management techniques) were recorded (for more informations about data, see data paper being published).
+    These datasets can be used to validate existing cropland and crop types/practices maps in the tropics, but also, to assess the performances and the robustness of classification methods of cropland and crop types/practices in a large range of Southern farming systems.
+
+    Citation: Jolivot, Audrey; Lebourgeois, Valentine; Ameline, Mael; Andriamanga, Valerie; Bellon, Beatriz; Castets, M
+    athieu; Crespin-Boucaud, Arthur; Defourny, Pierre; Diaz, Santiana; Dieye, Mohamadou; Dupuy, Stephane; Ferraz, Rodrigo;
+    Gaetano, Raffaele; Gely, Marie; Jahel, Camille; Kabore, Bertin; Lelong, Camille; Le Maire, Guerric; Leroux, Louise;
+    Lo Seen, Danny; Muthoni, Martha; Ndao, Babacar; Newby, Terry; De Oliveira Santos, Cecilia Lira Melo; Rasoamalala, Eloise;
+    Simoes, Margareth; Thiaw, Ibrahima; Timmermans, Alice; Tran, Annelise; Begue, Agnes, 2021,
+    "Harmonized in situ JECAM datasets for agricultural land use mapping and monitoring in tropical countries",
+    https://doi.org/10.18167/DVN1/P7OLAP, CIRAD Dataverse, V4
+    """
+    providers = [
+        {
+            "name": "Centre de coopération Internationale en Recherche Agronomique pour le Développement (Cirad)",
+            "url": "http://www.cirad.fr",
+            "roles": ["producer", "licensor"],
+        }
+    ]
+    attribution = "JECAM SIRAD, https://doi.org/10.18167/DVN1/P7OLAP"
+    license = "CC-BY-4.0"
+    columns = {
+        "geometry": "geometry",
+        "Id": "id",
+        "Area_ha": "area",
+        "AcquiDate": "determination_datetime",
+        "Country": "country_name",
+        "SiteName": "site_name",
+        "CropType1": "crop:name",
+        "Irrigated": "irrigated",
+    }
+    column_additions = {
+        "crop:code_list": "https://fiboa.org/code/jecam/crop.csv",
+    }
+    missing_schemas = {
+        "properties": {
+            "country_name": {"type": "string"},
+            "site_name": {"type": "string"},
+            "crop:name": {"type": "string"},
+            "crop:code_list": {"type": "string"},
+            "irrigated": {"type": "boolean"},
+        }
+    }
+
+    def migrate(self, gdf) -> gpd.GeoDataFrame:
+        gdf = super().migrate(gdf)
+        # Pick some CRS for missing area's, 32631 (central afrika) or 3857
+        gdf.loc[gdf["Area_ha"] == 0, "Area_ha"] = gdf.to_crs(32631).area / 10000
```

Comment:
Didn't we centralize this? We have the improve command and I think we did decide to not implement this in the converters individually, right?

Reply:
Yes, agreed. However, the dataset is invalid (`fiboa validate output.parquet --data`) because the fiboa schema forbids `area==0`. So I can leave the area column out (throwing away the area-data that's correct in most rows) or fill the 0-areas with a geometry-area calculation....

Reply:
I'm not sure if `Null/None/NaN` is allowed...

Reply:
null (Python None) is allowed.